File: day1/day1.jl
function getinput()
    nums = readlines("input.txt")
    nums = [parse(Int, x) for x in nums]
    return nums
end

function part1()
    nums = getinput()
    len = length(nums)
    global_index = 1

    while global_index < len
        curr_num = nums[global_index]
        curr_index = global_index + 1
        
        while curr_index < len          
            if curr_num + nums[curr_index] == 2020
                return curr_num * nums[curr_index]
            end
            
            curr_index = curr_index + 1
        end
        
        global_index = global_index + 1
    end
    
    return nothing
end

function part2()
    nums = getinput()
    len = length(nums)
    global_index = 1

    while global_index < len
        curr_num = nums[global_index]
        i = global_index + 1

        while i < len
            j = i + 1
            while j < len
                if curr_num + nums[i] + nums[j] == 2020
                    return curr_num * nums[i] * nums[j]
                end
                
                j = j + 1
            end

            i = i + 1
        end
         
        global_index = global_index + 1
    end

    return nothing
end

println(part1())

println(part2())

File: day1/day1.hs
main :: IO ()
    
main = do
    -- Read input
    input <- lines <$> readFile "input.txt"

     -- Convert to Int
    let nums = map read input :: [Int]

    let p1_result = part1 nums
    print p1_result

    let p2_result = part2 nums
    print p2_result

part1 :: [Int] -> Int
part1 nums = head [a*b | a <- nums, b <- nums, a + b == 2020]

part2 :: [Int] -> Int
part2 nums = head [a*b*c | a <- nums, b <- nums, c <- nums, a + b + c == 2020]
